core/sail/lucene/src/main/java/org/eclipse/rdf4j/sail/lucene/impl/LuceneIndex.java

```diff
@@ -912,20 +921,21 @@ public synchronized TopDocs search(Query query) throws IOException {
 		return getIndexSearcher().search(query, nDocs);
 	}
 
-	private QueryParser getQueryParser(IRI propertyURI) {
+	private QueryParser getQueryParser(final IRI propertyURI) {
+		final String fieldName;
 		// check out which query parser to use, based on the given property URI
-		if (propertyURI == null)
-		// if we have no property given, we create a default query parser
-		// which
-		// has the TEXT_FIELD_NAME as the default field
-		{
-			return new QueryParser(SearchFields.TEXT_FIELD_NAME, this.queryAnalyzer);
-		} else
-		// otherwise we create a query parser that has the given property as
-		// the default field
-		{
-			return new QueryParser(SearchFields.getPropertyField(propertyURI), this.queryAnalyzer);
+		if (propertyURI == null) {
+			// if we have no property given, we create a default query parser which has the TEXT_FIELD_NAME as the
+			// default field
+			fieldName = SearchFields.TEXT_FIELD_NAME;
+		} else {
+			// otherwise we create a query parser that has the given property as the default field
+			fieldName = SearchFields.getPropertyField(propertyURI);
 		}
+
+		final QueryParser queryParser = new QueryParser(fieldName, queryAnalyzer);
+		queryParser.setFuzzyPrefixLength(fuzzyPrefixLength);
+		return queryParser;
 	}
```

Comment:
Could you remove the `final` keywords here and in the other methods you've added? For these short and simple methods I feel they add more clutter than the safety and intent that they convey. For longer and more complex methods there is definitely a use case.